tutorials/week4-react.md

```diff
@@ -0,0 +1,689 @@
+---
+layout: page
+title: React Tutorial
+permalink: /tutorials/week4-react-basics
+parent: Tutorials
+nav_order: 6
+---
+
+# React Basics
+This tutorial covers the basic concepts of react. By the end of this tutorial, you will be able to create a new react app, understand the basic concepts of react such as states and props, understand React hooks and handling events.
+
+-   [React Basics](#react-basics)
+-   [Creating a new NextJs App](#creating-a-new-next-js-app)
+-   [Understanding a React App](#react-component)
+    -   [Components](#react-component)
+    -   [Template For Structure of Function Component](#template-for-structure-of-function-component)
+    -   [Props](#props)
+    -   [State](#state)
+    -   [Communicating between Components](#communication-between-components)
+        - [Parent to Child](#parent-to-child-component-communication)
+        - [Child to Parent](#child-to-parent-component-communication)
+    -   [Handling Events](#handling-events)
+-   [React Hooks](#react-hooks)
+    -   [UseState](#usestate)
+        - [Updating Arrays in State](#updating-arrays-in-state)
+    -   [UseEffect](#useeffect)
+        - [Object Dependencies in UseEffect](#object-dependencies-in-useEffect)
+
+
+## Creating a New Next Js App
+
+Let's use **npx** and **create-next-app** to create a new Next.Js project 
+
+-   **npx** stands for Node Package Execute (Part of the npm package since version 5.2). It is a runner that can execute any package that you want from the npm registry without even installing that package. In this case npx tool temporarily installs the create-next-app npm package and uses it to create our project.
+
+
+- Please check the following reference to the Next.Js boiler plate and getting started with framework 
+[Next.Js official documentation](https://nextjs.org/learn/react-foundations)
+
```

Comment:
I think we can one additional point for students, as this is common question.

- Before beginning the installation, please ensure that your Node.js version is 18 or higher, as Next.js is only supported on these versions. Use `node --version` command to check.